sdk/couchbase/tests/common/doc_generation.rs

```diff
@@ -108,6 +109,62 @@ pub async fn import_sample_beer_dataset(
     results
 }
 
+pub async fn import_projection_doc(collection: &Collection) -> String {
+    let content = read_to_string("tests/testdata/projection_doc.json").unwrap();
+    let doc: serde_json::Value = serde_json::from_str(content.as_str()).unwrap();
+
+    let key = new_key();
+    collection.upsert(key.clone(), &doc, None).await.unwrap();
+    key
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, PartialOrd, Default)]
```

Comment:
Do all of these throughout this JSON type need to be optional fields? Certainly name is always set.